buildtest/tools/buildsystem/singlesource.py

```diff
@@ -381,21 +383,26 @@ def __init__(self, file, lmod_collection, buildtest_collection):
 
         if "mpi" in self.test_yaml.keys():
             self.mpi = self.test_yaml["mpi"]
+
         if "moduleload" in self.test_yaml.keys():
             self.moduleload_check = True
             if "lmod_collection" in self.test_yaml["moduleload"].keys():
                 self.lmod_collection = self.test_yaml["moduleload"]["lmod_collection"]
 
         # self.srcfile = os.path.join(self.srcdir, self.test_yaml["program"]["source"])
         self.srcfile = self.test_yaml["program"]["source"]
-        self.execname = "%s.%s.exe" % (
-            os.path.basename(file),
-            hex(random.getrandbits(32)),
-        )
+
+        # Generate a common hex to identify and link exec and script file
+        randhex = hex(random.getrandbits(32))
+
+        self.execname = "%s.%s.exec" % (os.path.basename(config_file), randhex,)
```

Comment:
the comma after ``randhex,``  is not needed. 

I think you meant the following.

```
self.execname = "%s.%s.exec" % (os.path.basename(config_file), randhex)
```

Reply:
I think this was added by black, it automatically adds commas after the last item of the list. There should be no change to functionality. I can try removing it and running a commit again to be absolutely sure.

Reply:
This is fixed.